sklearn/manifold/tests/test_t_sne.py

```diff
@@ -256,13 +260,13 @@ def grad(params):
     assert_almost_equal(check_grad(fun, grad, X_embedded.ravel()), 0.0, decimal=5)
 
 
-def test_trustworthiness():
+def test_trustworthiness(global_dtype):
     # Test trustworthiness score.
     random_state = check_random_state(0)
 
     # Affine transformation
-    X = random_state.randn(100, 2)
-    assert trustworthiness(X, 5.0 + X / 10.0) == 1.0
+    X = random_state.randn(100, 2).astype(global_dtype)
+    assert trustworthiness(X, 5.0 + X / 10.0) == pytest.approx(1.0)
 
     # Randomly shuffled
     X = np.arange(100).reshape(-1, 1)
```

Comment:
I assume that we could also use `global_dtype` for this test.

Reply:
... and the same below and change `assert_almost_equal` by `assert ... == pytest.approx(...)`